File: backend/app/models/content_brief.py
import uuid
from datetime import date

from sqlalchemy import Date, Enum, ForeignKey, Integer, String, Text
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.models.base import Base, TimestampMixin, UUIDPrimaryKeyMixin


class ContentBrief(Base, UUIDPrimaryKeyMixin, TimestampMixin):
    __tablename__ = "content_briefs"

    organization_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True),
        ForeignKey("organizations.id", ondelete="CASCADE"),
        nullable=False,
    )
    client_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True),
        ForeignKey("clients.id", ondelete="CASCADE"),
        nullable=False,
    )
    name: Mapped[str] = mapped_column(String(255), nullable=False)
    content_type: Mapped[str] = mapped_column(
        Enum("feed", "story", "carousel", name="content_type_brief"),
        nullable=False,
        default="feed",
    )
    platform: Mapped[str] = mapped_column(
        Enum("instagram", "tiktok", "facebook", "twitter", "linkedin", name="platform_type"),
        nullable=False,
        default="instagram",
    )
    deadline_date: Mapped[date | None] = mapped_column(Date, nullable=True)
    status: Mapped[str] = mapped_column(
        Enum(
            "draft", "in_progress", "generated", "in_review", "approved", "published",
            name="brief_status",
        ),
        nullable=False,
        default="draft",
    )

    slides: Mapped[list["BriefSlide"]] = relationship(
        "BriefSlide",
        cascade="all, delete-orphan",
        order_by="BriefSlide.slide_number",
    )


class BriefSlide(Base, UUIDPrimaryKeyMixin):
    __tablename__ = "brief_slides"

    brief_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True),
        ForeignKey("content_briefs.id", ondelete="CASCADE"),
        nullable=False,
    )
    slide_title: Mapped[str] = mapped_column(String(255), nullable=False)
    brief_text: Mapped[str] = mapped_column(Text, nullable=False)
    notes: Mapped[str | None] = mapped_column(Text, nullable=True)
    slide_number: Mapped[int] = mapped_column(Integer, nullable=False, default=1)

File: frontend/src/types/content-brief.ts
export interface ContentBrief {
  id: string
  organization_id: string
  client_id: string
  name: string
  content_type: ContentType
  platform: Platform
  deadline_date: string | null
  status: ContentStatus
  slides: BriefSlide[]
  created_at: string | null
}

export interface BriefSlide {
  id: string
  slide_title: string
  brief_text: string
  notes: string | null
  slide_number: number
}

export type Platform = "instagram" | "tiktok" | "facebook" | "twitter" | "linkedin"
export type ContentType = "feed" | "story" | "carousel"
export type ContentStatus =
  | "draft"
  | "in_progress"
  | "generated"
  | "in_review"
  | "approved"
  | "published"

export const PLATFORM_LABELS: Record<Platform, string> = {
  instagram: "Instagram",
  tiktok: "TikTok",
  facebook: "Facebook",
  twitter: "Twitter/X",
  linkedin: "LinkedIn",
}

export const PLATFORM_BADGE_VARIANTS: Record<Platform, string> = {
  instagram: "bg-pink-100 text-pink-700",
  tiktok: "bg-neutral-900 text-white",
  facebook: "bg-blue-100 text-blue-700",
  twitter: "bg-slate-100 text-slate-700",
  linkedin: "bg-cyan-100 text-cyan-700",
}

export const CONTENT_TYPE_LABELS: Record<ContentType, string> = {
  feed: "Feed",
  story: "Story",
  carousel: "Carousel",
}

export const CONTENT_STATUS_LABELS: Record<ContentStatus, string> = {
  draft: "Draft",
  in_progress: "In Progress",
  generated: "Generated",
  in_review: "In Review",
  approved: "Approved",
  published: "Published",
}
